diff --git a/librz/bin/mangling/cxx/libiberty.h b/librz/bin/mangling/cxx/libiberty.h
index e2c3436010ad..cdf56c87abbd 100644
--- a/librz/bin/mangling/cxx/libiberty.h
+++ b/librz/bin/mangling/cxx/libiberty.h
@@ -622,14 +622,18 @@ extern int pwait (int, int *, int);
 /* Like sprintf but provides a pointer to malloc'd storage, which must
    be freed by the caller.  */
 
-extern int asprintf (char **, const char *, ...) ATTRIBUTE_PRINTF_2;
+extern int asprintf (char **__restrict __ptr,
+            const char *__restrict __fmt, ...)
+      __THROWNL __attribute__ ((__format__ (__printf__, 2, 3))) __wur;
 #endif
 
 #if !HAVE_DECL_VASPRINTF
 /* Like vsprintf but provides a pointer to malloc'd storage, which
    must be freed by the caller.  */
 
-extern int vasprintf (char **, const char *, va_list) ATTRIBUTE_PRINTF(2,0);
+extern int vasprintf (char **__restrict __ptr, const char *__restrict __f,
+               __gnuc_va_list __arg)
+      __THROWNL __attribute__ ((__format__ (__printf__, 2, 0))) __wur;
 #endif
 
 #if defined(HAVE_DECL_SNPRINTF) && !HAVE_DECL_SNPRINTF
diff --git a/shlr/rizin-shell-parser/grammar.js b/shlr/rizin-shell-parser/grammar.js
index f444b559022d..580e295d4b98 100644
--- a/shlr/rizin-shell-parser/grammar.js
+++ b/shlr/rizin-shell-parser/grammar.js
@@ -58,7 +58,7 @@ module.exports = grammar({
     $._spec_sep,
   ],
 
-  inline: ($) => [$.stmt_delimiter, $.stmt_delimiter_singleline, $._comment],
+  inline: ($) => [$.stmt_delimiter, $.stmt_delimiter_singleline],
 
   rules: {
     statements: ($) =>
