From 7c837e4a043e1f8c24b94ce19b32f30cc6c04435 Mon Sep 17 00:00:00 2001
From: Peter Kasting <pkasting@chromium.org>
Date: Mon, 26 Jul 2021 04:15:53 -0700
Subject: [PATCH] Fix an instance of -Wunused-but-set-variable.

Bug: chromium:1203071
---
 asm/preproc.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/asm/preproc.c b/asm/preproc.c
index d1b9b31b..3a0db1ac 100644
--- a/asm/preproc.c
+++ b/asm/preproc.c
@@ -5574,7 +5574,6 @@ static SMacro *expand_one_smacro(Token ***tpp)
         int brackets = 0;
         int paren;
         bool bracketed = false;
-        bool bad_bracket = false;
         enum sparmflags flags;
 
         if (m->params[m->nparam-1].flags & SPARM_GREEDY)
@@ -5655,8 +5654,6 @@ static SMacro *expand_one_smacro(Token ***tpp)
             if (!skip) {
                 Token *t;
 
-                bad_bracket |= bracketed && !brackets;
-
                 if (white) {
                     *pep = t = new_White(NULL);
                     pep = &t->next;
