diff --git a/libdwfl/linux-kernel-modules.c b/libdwfl/linux-kernel-modules.c
index c0f8dfa4aef2..44cb4f445d84 100644
--- a/libdwfl/linux-kernel-modules.c
+++ b/libdwfl/linux-kernel-modules.c
@@ -1015,7 +1015,7 @@ dwfl_linux_kernel_report_modules (Dwfl *dwfl)
      from other whitespace so as to take the optional word following the
      address but always stop at the end of the line.  */
   while (getline (&line, &linesz, f) > 0
-	 && sscanf (line, "%128s %lu %*s %*s %*s %" PRIx64 " %*s\n",
+	 && sscanf (line, "%127s %lu %*s %*s %*s %" PRIx64 " %*s\n",
 		    modname, &modsz, &modaddr) == 3)
     {
       Dwfl_Module *mod = INTUSE(dwfl_report_module) (dwfl, modname,
