From 26f848d4235f796878ffbc4780a6283528c00d93 Mon Sep 17 00:00:00 2001
From: Henri Chain <henri.chain@enioka.com>
Date: Tue, 14 Sep 2021 21:58:28 +0200
Subject: [PATCH] fix systemd detection D-Bus calls

---
 startkde/startplasma.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/startkde/startplasma.cpp b/startkde/startplasma.cpp
index 699d048e4..8d0b0ba89 100644
--- a/startkde/startplasma.cpp
+++ b/startkde/startplasma.cpp
@@ -439,17 +439,19 @@ void resetSystemdFailedUnits()
 
 bool hasSystemdService(const QString &serviceName)
 {
+    qDBusRegisterMetaType<QPair<QString, QString>>();
+    qDBusRegisterMetaType<QList<QPair<QString, QString>>>();
     auto msg = QDBusMessage::createMethodCall(QStringLiteral("org.freedesktop.systemd1"),
                                               QStringLiteral("/org/freedesktop/systemd1"),
                                               QStringLiteral("org.freedesktop.systemd1.Manager"),
                                               QStringLiteral("ListUnitFilesByPatterns"));
     msg << QStringList({QStringLiteral("enabled"), QStringLiteral("static")}) << QStringList({serviceName});
-    auto reply = QDBusConnection::sessionBus().call(msg);
-    if (reply.type() == QDBusMessage::ErrorMessage) {
+    QDBusReply<QList<QPair<QString, QString>>> reply = QDBusConnection::sessionBus().call(msg);
+    if (!reply.isValid()) {
         return false;
     }
     // if we have a service returned then it must have found it
-    return !reply.arguments().isEmpty();
+    return !reply.value().isEmpty();
 }
 
 bool useSystemdBoot()
